src/domain/synthetics/sidecarOrders/utils.ts

```diff
@@ -0,0 +1,239 @@
+  }
+): T {
+  let sizeError: string | null = null;
+  let priceError: string | null = null;
+  let percentageError: string | null = null;
+
+  const inputPrice = entry.price.value;
+
+  if (inputPrice?.gt(0)) {
+    if (markPrice) {
+      if (type === "limit") {
+        priceError = isLong
+          ? inputPrice.gt(markPrice)
+            ? t`Price above Mark Price.`
+            : priceError
+          : inputPrice.lt(markPrice)
+          ? t`Price below Mark Price.`
+          : priceError;
+      }
+    }
+
+    if (!isExistingLimits && liqPrice) {
+      if (type === "sl") {
+        priceError = isLong
+          ? inputPrice.lt(liqPrice)
+            ? t`Price below Liq. Price.`
+            : priceError
+          : inputPrice.gt(liqPrice)
+          ? t`Price above Liq. Price.`
+          : priceError;
+      }
+    }
+
+    if (isExistingPosition || !isLimit) {
+      if (markPrice) {
+        if (type === "tp") {
+          priceError = isLong
+            ? inputPrice.lt(markPrice)
+              ? t`Price below Mark Price.`
+              : priceError
+            : inputPrice.gt(markPrice)
+            ? t`Price above Mark Price.`
+            : priceError;
+        }
+
+        if (type === "sl") {
+          priceError = isLong
+            ? inputPrice.gt(markPrice)
+              ? t`Price above Mark Price.`
+              : priceError
+            : inputPrice.lt(markPrice)
+            ? t`Price below Mark Price.`
+            : priceError;
+        }
+      }
+    } else {
+      if (isExistingLimits) {
+        if (maxLimitTrigerPrice && minLimitTrigerPrice) {
+          if (type === "tp") {
+            priceError = isLong
+              ? inputPrice.lt(maxLimitTrigerPrice)
+                ? t`Price below highest Limit Price.`
+                : priceError
+              : inputPrice.gt(minLimitTrigerPrice)
+              ? t`Price above lowes Limit Price.`
+              : priceError;
+          }
+
+          if (type === "sl") {
+            priceError = isLong
+              ? inputPrice.gt(maxLimitTrigerPrice)
+                ? t`Price above highest Limit Price.`
+                : priceError
+              : inputPrice.lt(minLimitTrigerPrice)
+              ? t`Price below lowes Limit Price.`
+              : priceError;
+          }
+        }
+      } else {
+        if (triggerPrice) {
+          if (type === "tp") {
+            priceError = isLong
+              ? inputPrice.lt(triggerPrice)
+                ? t`Price below Limit Price.`
+                : priceError
+              : inputPrice.gt(triggerPrice)
+              ? t`Price above Limit Price.`
+              : priceError;
+          }
+
+          if (type === "sl") {
+            priceError = isLong
+              ? inputPrice.gt(triggerPrice)
+                ? t`Price above Limit Price.`
+                : priceError
+              : inputPrice.lt(triggerPrice)
+              ? t`Price below Limit Price.`
+              : priceError;
```

Comment:
these conditions looks very repetitive but don't have a strong idea what to do with it

Reply:
Yep, it's a tricky place, hope it's a bit more readable now